tools/png2c.py

```diff
@@ -296,16 +296,12 @@ def do_palette(im, desc):
     param = parse(desc, ('name', str), ('colors', int))
 
     name = param['name']
-    has_colors = param['colors']
+    colors = param['colors']
 
     pal = im.getpalette()
-    colors = im.getextrema()[1] + 1
```

Comment:
I would have preferred to keep the "actual palette versus required palette" check, using the palette from the PNG file instead of the histogram of the used colors. However, using `im.getpalette()` I always got 256 color entries, even if the PNG only has 32 palette entries.

Reply:
Have you tried to run `optipng` over the image? Usually that helps with images that have 256 but only subset used.